### DISCUSSION TITLE
Improve warning of incorrect configured relationship in m2m interface

### DISCUSSION THREAD

**Discussion by christian-hackyourshack** [directus/directus#13490] [Feature Requests]
### Describe the Bug

It took me quite some time to figure this one out:

I have an M2M relation configured by an O2M-M2O pair of relations. The junction table is created manually and has a few attributes on the relation. Imagine a recipe referencing ingredients: The reference has an amount, a unit, a comment and can be sorted.

I do not know the history of the table in detail, but it was configured to be O2M on the recipe side (field ingredients), but had the junction_field set in the `directus_relations` table, which wasn't a problem in 9.5.2 but after migrating to 9.11 it showed up as "The relationship hasn't been configured correctly".

I couldn't find anything in the logs or any further hint what was upsetting the app, so I had to analyze the SQL (Postgres) until I found this issue.

My proposal:

- provide more detailed error information in the logs in the error dialog, wherever
- reduce the configuration redundancy (doesn't the junction_field obviously imply M2M?)
- make the configuration parameters more obvious, I had to finally reset the field in the SQL, because I could not find it in the UI (my choice for Directus was, because I could simply take my DB as is and quickly configure some web forms on top of it, but without deleting a data table I could not get rid of this configuration parameter from the configuration UI).

### To Reproduce

Manually set the junction_field in the directus_relation for an O2M relation. (I did not do this manually, but inherited it from some older configuration settings and switching from an M2M to a manually tweaked O2M-M2O relation.)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
